neo4j/hooks.go

```diff
@@ -0,0 +1,7 @@
+package neo4j
+
+import "net/url"
+
+type PreQueryHook = func(cypher string, params map[string]interface{}, target url.URL, config *TransactionConfig)
```

Comment:
Does it make sense to implement just one or the other? If not an interface should be used.
Passing config as a pointer allows for modifications which would be another way of modifiying the tx, intended or unintended? Why would instrumentation hooks need to modify the config?
No need for aliasing in this case.

Reply:
Not perfect, but intended, since I'd have to change the context (actually overriding it with a new instance, which now contains the current span for the execution (to retrieve it in the second hook and finish it)